crates/web-sys/src/lib.rs

```diff
@@ -17,4 +17,15 @@ extern crate wasm_bindgen;
 extern crate js_sys;
 use js_sys::Object;
 
+#[cfg(feature = "Window")]
+pub fn window() -> Option<Window> {
```

Comment:
Yeah I was originally wondering that as well, but I'm somewhat wary to do that because we get things like:

```
crates/web-sys/webidls/enabled/ServiceWorkerGlobalScope.webidl:[Global=(Worker,ServiceWorker),
crates/web-sys/webidls/enabled/ServiceWorkerGlobalScope.webidl- Exposed=ServiceWorker]
crates/web-sys/webidls/enabled/ServiceWorkerGlobalScope.webidl-interface ServiceWorkerGlobalScope : WorkerGlobalScope {
--
crates/web-sys/webidls/enabled/AudioWorkletGlobalScope.webidl:[Global=(Worklet,AudioWorklet),Exposed=AudioWorklet]
crates/web-sys/webidls/enabled/AudioWorkletGlobalScope.webidl-interface AudioWorkletGlobalScope : WorkletGlobalScope {
crates/web-sys/webidls/enabled/AudioWorkletGlobalScope.webidl-    void registerProcessor (DOMString name, VoidFunction processorCtor);
--
crates/web-sys/webidls/enabled/PaintWorkletGlobalScope.webidl:[Global=(Worklet,PaintWorklet),Exposed=PaintWorklet]
crates/web-sys/webidls/enabled/PaintWorkletGlobalScope.webidl-interface PaintWorkletGlobalScope : WorkletGlobalScope {
crates/web-sys/webidls/enabled/PaintWorkletGlobalScope.webidl-    void registerPaint(DOMString name, VoidFunction paintCtor);
--
crates/web-sys/webidls/enabled/Window.webidl:[Global=Window,
crates/web-sys/webidls/enabled/Window.webidl- Exposed=Window,
crates/web-sys/webidls/enabled/Window.webidl- LegacyUnenumerableNamedProperties]
--
crates/web-sys/webidls/enabled/DedicatedWorkerGlobalScope.webidl:[Global=(Worker,DedicatedWorker),
crates/web-sys/webidls/enabled/DedicatedWorkerGlobalScope.webidl- Exposed=DedicatedWorker]
crates/web-sys/webidls/enabled/DedicatedWorkerGlobalScope.webidl-interface DedicatedWorkerGlobalScope : WorkerGlobalScope {
--
crates/web-sys/webidls/enabled/WorkerDebuggerGlobalScope.webidl:[Global=(WorkerDebugger), Exposed=WorkerDebugger]
crates/web-sys/webidls/enabled/WorkerDebuggerGlobalScope.webidl-interface WorkerDebuggerGlobalScope : EventTarget {
crates/web-sys/webidls/enabled/WorkerDebuggerGlobalScope.webidl-  [Throws]
--
crates/web-sys/webidls/enabled/SharedWorkerGlobalScope.webidl:[Global=(Worker,SharedWorker),
crates/web-sys/webidls/enabled/SharedWorkerGlobalScope.webidl- Exposed=SharedWorker]
crates/web-sys/webidls/enabled/SharedWorkerGlobalScope.webidl-interface SharedWorkerGlobalScope : WorkerGlobalScope {
```

It looks like there's a lot of semi-weird `*GlobalScope` objects perhaps?

One option we could do though is to perhaps define a `self_` method which returns `self`, and then have other handwritten methods that cast this to the various scopes? I also figured that a top-level `window` accessor should at least be forwards compatible with an automatic solution!

Reply:
Yeah, I suppose it is fine to special case and hand-code globals right now.

If you are writing an audio worklet, you will want to get the audio worklet global scope... but almost no one is doing that (relative to people using window)

WorkerDebuggerGlobalScope can probably be removed -- I think that is only for the debugger.

I think a `fn self_() -> js_sys::Object` function would be useful regardless. :+1: